mava/systems/tf/maddpg/system.py

```diff
@@ -227,10 +228,12 @@ def __init__(  # noqa
         if type(network_sampling_setup) is not list:
             if network_sampling_setup == enums.NetworkSampler.fixed_agent_networks:
                 # if no network_sampling_setup is fixed, use shared_weights to
-                # determine setup
+                # assign a single network to all agents of the same type
                 self._agent_net_keys = {
-                    agent: "network_0" if shared_weights else f"network_{i}"
-                    for i, agent in enumerate(agents)
+                    agent: f"""network_{agent.split("_")[0]}"""
```

Comment:
Why use triple quotes here? Will `f"network_{agent.split("_")[0]}"` not do?

Reply:
Triple quotes are needed due to the use of double quotes in "_". It is possible to use single quotes but the code formatter automatically converts ' to "
https://stackoverflow.com/questions/68461626/how-to-fix-unterminated-expression-in-f-string-missing-close-brace-in-python

Reply:
Triple quotes are needed due to the use of double quotes in "_". It is possible to use single quotes but the code formatter automatically converts ' to "
https://stackoverflow.com/questions/68461626/how-to-fix-unterminated-expression-in-f-string-missing-close-brace-in-python

Reply:
Ah I see, thanks 👍